From 391e60944c1ea383d3f9ca983c383c93672fc346 Mon Sep 17 00:00:00 2001
From: Ray Donnelly <mingw.android@gmail.com>
Date: Fri, 24 Jan 2014 18:22:19 +0000
Subject: [PATCH 17/28] Add GCC 4.8.2 hack cp -p as ln -s issues

When building a multilib libgcc on Windows, cp -p is used
but the semantics are quite different from those of ln -s
so we cd to the folder where the link will be created and
use LN_S (cp -p) without folder names.
---
 ...w-hack-around-cp-p-as-ln-s-issues-libgcc-s.patch | 21 +++++++++++++++++++++
 1 file changed, 21 insertions(+)
 create mode 100644 patches/gcc/4.8.2/160-mingw-hack-around-cp-p-as-ln-s-issues-libgcc-s.patch

diff --git a/patches/gcc/4.8.2/160-mingw-hack-around-cp-p-as-ln-s-issues-libgcc-s.patch b/patches/gcc/4.8.2/160-mingw-hack-around-cp-p-as-ln-s-issues-libgcc-s.patch
new file mode 100644
index 0000000..33f361d
--- /dev/null
+++ b/patches/gcc/4.8.2/160-mingw-hack-around-cp-p-as-ln-s-issues-libgcc-s.patch
@@ -0,0 +1,21 @@
+diff -urN gcc-4.8.2.orig/libgcc/config/t-slibgcc gcc-4.8.2/libgcc/config/t-slibgcc
+--- gcc-4.8.2.orig/libgcc/config/t-slibgcc	2013-02-04 19:06:20.000000000 +0000
++++ gcc-4.8.2/libgcc/config/t-slibgcc	2014-01-24 18:12:07.842866500 +0000
+@@ -27,9 +27,14 @@
+ SHLIB_DIR = @multilib_dir@
+ SHLIB_SLIBDIR_QUAL = @shlib_slibdir_qual@
+ SHLIB_LC = -lc
+-SHLIB_MAKE_SOLINK = $(LN_S) $(SHLIB_SONAME) $(SHLIB_DIR)/$(SHLIB_SOLINK)
+-SHLIB_INSTALL_SOLINK = $(LN_S) $(SHLIB_SONAME) \
+-	$(DESTDIR)$(slibdir)$(SHLIB_SLIBDIR_QUAL)/$(SHLIB_SOLINK)
++SHLIB_MAKE_SOLINK = SHLIB_MAKE_SOLINK_PWD=$$PWD && \
++	cd $(SHLIB_DIR) && \
++	$(LN_S) $(SHLIB_SONAME) $(SHLIB_SOLINK) && \
++	cd $$SHLIB_MAKE_SOLINK_PWD
++SHLIB_INSTALL_SOLINK = SHLIB_MAKE_SOLINK_PWD=$$PWD && \
++	cd $(DESTDIR)$(slibdir)$(SHLIB_SLIBDIR_QUAL) && \
++	$(LN_S) $(SHLIB_SONAME) $(SHLIB_SOLINK) && \
++	cd $$SHLIB_MAKE_SOLINK_PWD
+ 
+ SHLIB_LINK = $(CC) $(LIBGCC2_CFLAGS) -shared -nodefaultlibs \
+ 	$(SHLIB_LDFLAGS) \
-- 
1.8.5.3

